### ISSUE TITLE
MSYS2 toolchain no longer works out of the box

### ISSUE THREAD

**Issue by petrosagg** [lkl/linux#289]
In March MSYS2 rolled back the version of binutils to 2.15 that doesn't include the weak NT externals patches. https://github.com/Alexpux/MSYS2-packages/commit/660ffd3fd117a22f3dff71d80edf56ad4a188c26 

One can use the following workaround to install the previous version:

```bash
wget http://repo.msys2.org/msys/i686/binutils-2.26-1-i686.pkg.tar.xz
pacman -U binutils-2.26-1-i686.pkg.tar.xz
```

**Comment by Rondom** [comment]
I consider it a regression that they dropped the patches for weak externals. Have you tried filing a bug report when them?

**Comment by petrosagg** [comment]
No, I'll do that now. However I suspect the reason is that that the patches don't apply cleanly on binutils 2.25.1. I'm rebasing them now and I'll PR them to MSYS2.

**Comment by petrosagg** [comment]
I sent a PR. @tavip I had to do some minor changes to your patches in order for them to apply on 2.25.1. If could take a look at them it would be great since I'm not very familiar with the binutils codecase. https://github.com/Alexpux/MSYS2-packages/pull/785

**Comment by tavip** [comment]
The changes looks good to me, I've left a message in the PR too :) More importantly, if the LKL buid and tests pass, it should be fine, since it stresses this feature pretty well.

**Comment by Rondom** [comment]
The MinGW-version of the package is lacking the patch as well. Any takers? ;-)
https://github.com/Alexpux/MINGW-packages/tree/master/mingw-w64-binutils
